sklearn/metrics/tests/test_score_objects.py

```diff
@@ -526,3 +546,39 @@ def test_scoring_is_not_metric():
                          Ridge(), r2_score)
     assert_raises_regexp(ValueError, 'make_scorer', check_scoring,
                          KMeans(), cluster_module.adjusted_rand_score)
+
+
+def test_pass_classes():
+    # Test the various properties of pass_classes parameter added
+
+    X, y = make_blobs(random_state=0, centers=2)
+    clf = LogisticRegression(random_state=0)
+    clf.fit(X, y)
+
+    # copy_classes=None should not set labels
+    scorer = make_scorer(DummyScorerWithLabels, pass_classes=None)
+    assert_equal(scorer(clf, X, y), 0)
+
+    # by default, labels should be set
+    scorer = make_scorer(DummyScorerWithLabels)
+    assert_equal(scorer(clf, X, y), 1)
+
+    # by default, labels should be set
+    scorer = make_scorer(DummyScorerWithLabels, labels=[1, 2, 3])
+    expected_msg = ("`estimator classes=[0 1]` is not the"
+                    "superset of `scorer labels=[1, 2, 3]`")
+    assert_raise_message(ValueError, expected_msg,
```

Comment:
Add a test for the subset warning, make a scorer with ``allow_label_subset`` and test that it doesn't warn and has the same result.

Reply:
won't that be covered in the test of metrics as that check happens in metrics?